app/parse.py

```diff
@@ -8,5 +11,26 @@
     duration: str
 
 
+def parse_single_course(course_element: Tag) -> Course:
+    name = course_element.select_one("h3").text.strip()
+    short_description = course_element.select_one(
+        ".typography_landingTextMain__Rc8BD"
+    ).text.strip()
+    duration = course_element.select(
+        ".typography_landingTextMain__Rc8BD span"
+    )[-1].text.strip()
+
+    return Course(name, short_description, duration)
+
+
 def get_all_courses() -> list[Course]:
-    pass
+    content = requests.get("https://mate.academy/")
+    content.raise_for_status()
+
+    soup = BeautifulSoup(content.text, "html.parser")
+    course_elements = soup.select(".ProfessionCard_cardWrapper__2Q8_V")
```

Comment:
Verify that the selector '.ProfessionCard_cardWrapper__2Q8_V' accurately targets all course elements. Changes in the webpage's HTML structure could affect the parsing results.